### PR TITLE
Fix Scoped Storage (Android API 30 Requirement)

### PR DESCRIPTION
**Pull request by davidofwatkins**
This contains the same changes/fixes from https://github.com/apache/cordova-plugin-media-capture/pull/215, but rebased against the latest upstream `master`. I've also addressed some of the open comments in that PR.

Ideally that will get merged soon and we can throw out this fork.

### CODE REVIEW

File: plugin.xml [hunk 1]
```diff
@@ -19,7 +19,7 @@
 -->
 
 <plugin xmlns="http://apache.org/cordova/ns/plugins/1.0"
-xmlns:android="http://schemas.android.com/apk/res/android"
+xmlns:android="http://schemas.android.com/apk/res/andraoid"
```
Comment (jorgetinoco) [inline plugin.xml]: is this change intentional?

File: src/android/Capture.java [hunk 2]
```diff
@@ -266,7 +278,18 @@ private void captureAudio(Request req) {
 
         try {
             Intent intent = new Intent(android.provider.MediaStore.Audio.Media.RECORD_SOUND_ACTION);
-            this.cordova.startActivityForResult((CordovaPlugin) this, intent, req.requestCode);
+
+            String timeStamp = new SimpleDateFormat("yyyyMMddHHmmssSSS").format(new Date());
+              String fileName = "AUDIO_" + timeStamp + ".wav";
```
Comment (jorgetinoco) [inline src/android/Capture.java]: just a suggestion, I would extract this way of getting a file to a function, something like
```java
private static final String DATE_FORMAT = "yyyyMMddHHmmssSSS";
private File getFile(String type, string extension) {
  String timeStamp = new SimpleDateFormat(DATE_FORMAT).format(new Date());
  String filename = String.format("$s_%s.%s",type, timeStamp, extension);
  return new File(getTempDirectoryPath(), filename);
}
```

File: src/android/Capture.java [hunk 3]
```diff
@@ -413,45 +444,31 @@ public void onImageActivityResult(Request req) {
...
-            req.results.put(createMediaFile(data));
 
-            if (req.results.length() >= req.limit) {
-                // Send Uri back to JavaScript for viewing video
-                pendingRequests.resolveWithSuccess(req);
-            } else {
-                // still need to capture more video clips
-                captureVideo(req);
-            }
+        if (req.results.length() >= req.limit) {
```
Comment (jorgetinoco) [inline src/android/Capture.java:450]: not sure if you would need to check for `req.results != null && req.results.length....`

File: src/android/Capture.java [hunk 4]
```diff
@@ -294,13 +317,16 @@ private void captureImage(Request req) {
 
         Intent intent = new Intent(android.provider.MediaStore.ACTION_IMAGE_CAPTURE);
 
-        ContentResolver contentResolver = this.cordova.getActivity().getContentResolver();
-        ContentValues cv = new ContentValues();
-        cv.put(MediaStore.Images.Media.MIME_TYPE, IMAGE_JPEG);
-        imageUri = contentResolver.insert(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, cv);
-        LOG.d(LOG_TAG, "Taking a picture and saving to: " + imageUri.toString());
+        String timeStamp = new SimpleDateFormat("yyyyMMddHHmmssSSS").format(new Date());
```
Comment (felixdelacruzt) [inline src/android/Capture.java]: the function suggested by @jorgetinoco above, can be reuse here too.

File: src/android/Capture.java [hunk 5]
```diff
@@ -317,6 +343,15 @@ private void captureVideo(Request req) {
         if (isMissingCameraPermissions(req)) return;
 
         Intent intent = new Intent(android.provider.MediaStore.ACTION_VIDEO_CAPTURE);
+            String timeStamp = new SimpleDateFormat("yyyyMMddHHmmssSSS").format(new Date());
```
Comment (felixdelacruzt) [inline src/android/Capture.java]: same comment as above ^

File: src/android/Capture.java (hunk 3 above)
Comment (davidofwatkins) [reply src/android/Capture.java:450]: I think this is okay since it was this way before; it was just moved.

File: plugin.xml (hunk 1 above)
Comment (davidofwatkins) [reply plugin.xml]: Oops! Nope, not sure how this snuck in here 😅 Thanks!

File: src/android/Capture.java (hunk 2 above)
Comment (davidofwatkins) [reply src/android/Capture.java]: Good call, updated!

**Review by felixdelacruzt — APPROVED** [review]
LGTM 👍

**Review by jorgetinoco — APPROVED** [review]
LGTM!

**Comment by davidofwatkins** [comment]
Thanks for the reviews @jorgetinoco and @felixdelacruzt!
